src/aerospike_vector_search/aio/internal/channel_provider.py

```
@@ -69,42 +65,31 @@ def __init__(
         # Configure token manager for async operation
         self._token_manager.configure_async(self._auth_tending_lock)
 
-        # initializes authentication tending
-        self._auth_task = asyncio.create_task(
-            self._tend_token()
-        )
-
         # initializes client tending processes
-        asyncio.create_task(self._tend())
+        asyncio.create_task(self._start_tending())
 
     async def _is_ready(self):
         # Wait 1 round of cluster tending, auth token initialization, and server client compatibility verification
         await self._ready.wait()
 
-        # This propogates any fatal/unexpected errors from client initialization/tending to the client.
+        # This propagates any fatal/unexpected errors from client initialization/tending to the client.
         # Raising errors in a task does not deliver this error information to users
         if self._tend_exception:
             raise self._tend_exception
 
-    async def _tend(self):
+    async def _start_tending(self):
 
         try:
-            # Wait for auth task if it exists
-            if self._auth_task:
-                await self._auth_task
-            else:
-                logger.error("Auth task not initialized")
-                e = types.AVSClientError(message="Auth task not initialized")
-                self._tend_exception = e
-                raise e
+            auth_stub = self._get_auth_stub()
+            await self._token_manager.refresh_token_async(auth_stub)
```

Since _auth_task is removed, this refresh will error will not be propagated to the user.  As long as the user gets an error message that notifies them that their jwt credentials are invalid, this should be fine.

However, this is not in alignment with the sync client, which will propagate the error to the user on the first tend iteration, but will only log the error any subsequent iteration.

---

_start_tending wraps refresh_token_async in a try except that sets _tend_exception, which is bubbled up. When refresh_token_async is called it refreshes the token immediately, outside of an asyncio.Task, so any exceptions will be raised inside _start_tend which will set the _tend_exception. After this immediate refresh, refresh_token_async schedules itself to be called in an asyncio.Task later, based on my understanding, those exceptions will be swallowed but those from the first run will be raised.